str/trtools/merge_str_prep.py

```diff
@@ -52,57 +54,66 @@ def main(
-    }
+    if sharded:
+        input_vcf_dict = {
+            id: [str(gspath) for gspath in to_path(f'{input_dir}/{id}').glob('*.vcf')]
+            for id in internal_wgs_ids
+        }
+
+    else:
+        input_vcf_dict = {
+            id: [os.path.join(input_dir, f'{id}_{caller}.vcf')]
+            for id in internal_wgs_ids
+        }
 
     for id in list(input_vcf_dict.keys()):
-        bcftools_job = b.new_job(name=f'{id} {caller} Files prep')
-        bcftools_job.image(BCFTOOLS_IMAGE)
-        bcftools_job.cpu(4)
-        bcftools_job.storage('15G')
-
-        vcf_input = b.read_input(input_vcf_dict[id])
-
-        if caller == 'eh':
-            bcftools_job.declare_resource_group(
-                vcf_sorted={
-                    'reheader.vcf.gz': '{root}.reheader.vcf.gz',
-                    'reheader.vcf.gz.tbi': '{root}.reheader.vcf.gz.tbi',
-                }
-            )
-            bcftools_job.command(
-                f"""
-
-                bcftools reheader -f {ref.fai} {vcf_input} | bcftools sort --temp-dir $BATCH_TMPDIR/ | bgzip -c >  {bcftools_job.vcf_sorted['reheader.vcf.gz']}
-
-                tabix -f -p vcf {bcftools_job.vcf_sorted['reheader.vcf.gz']}
-
-                """
-            )
-            # Output writing
-            output_path_eh = output_path(f'{id}_eh', 'analysis')
-            b.write_output(bcftools_job.vcf_sorted, output_path_eh)
-
-        else:
-            bcftools_job.declare_resource_group(
-                vcf_sorted={
-                    'vcf.gz': '{root}.vcf.gz',
-                    'vcf.gz.tbi': '{root}.vcf.gz.tbi',
-                }
-            )
-            bcftools_job.command(
-                f"""
-
-                bcftools sort {vcf_input} | bgzip -c  > {bcftools_job.vcf_sorted['vcf.gz']}
-
-                tabix -p vcf {bcftools_job.vcf_sorted['vcf.gz']}
-
-                """
-            )
-            # Output writing
-            output_path_gangstr = output_path(f'{id}_gangstr', 'analysis')
-            b.write_output(bcftools_job.vcf_sorted, output_path_gangstr)
+        for vcf_file in input_vcf_dict[id]:
+            vcf_input = b.read_input(vcf_file)
+            bcftools_job = b.new_job(name=f'{vcf_file} Files prep')
+            bcftools_job.image(BCFTOOLS_IMAGE)
+            bcftools_job.cpu(4)
+            bcftools_job.storage('15G')
+
+            if caller == 'eh':
+                bcftools_job.declare_resource_group(
+                    vcf_sorted={
+                        'reheader.vcf.gz': '{root}.reheader.vcf.gz',
+                        'reheader.vcf.gz.tbi': '{root}.reheader.vcf.gz.tbi',
+                    }
+                )
+                bcftools_job.command(
+                    f"""
+
+                    bcftools reheader -f {ref.fai} {vcf_input} | bcftools sort --temp-dir $BATCH_TMPDIR/ | bgzip -c >  {bcftools_job.vcf_sorted['reheader.vcf.gz']}
```

Comment:
I'm in favour of breaking these into different lines as it helps readability (in Python and Hail Batch)